src/server/util/rolesHelpers.js

```diff
@@ -0,0 +1,147 @@
+const { Pool } = require('pg');
+
+const { apiError, asyncMiddleware } = require('./api');
+const { sql } = require('./sqlHelpers');
+
+// Do we want our own pool or should this be shared with the user authentication pool?
+const pool = new Pool();
+
+const getUserInDatabase = async function(user) {
+    const client = await pool.connect();
+    const result = await client.query(
+        sql`SELECT * FROM users WHERE email = ${user} OR username = ${user};`
+    );
+    const userId = result.rows[0] ? result.rows[0].id : null; // Should we account for getting more than one user back?
```

Comment:
Nonblocking, if answer to first question is No: Is there a scenario when you've gotten back more than one user? If so, we should tighten up the query.

Reply:
the query should maybe just be `LIMIT 1`

Reply:
though honestly, i'd rather the API work from user ID (the number) instead of the username/email (who knows....)

Reply:
I haven't gotten more than one user, but I could imagine a scenario where we did get one if people create duplicate accounts. `LIMIT 1` would fix this query, though not the duplicate account issue. Maybe I'm overthinking it though, we do have helpers to make sure dupes aren't added. I'll add LIMIT 1 to this query and we can cross the dupe-account-bridge when/if we get to it.